### DISCUSSION TITLE
Plugin template tags in requests cause errors in Inso

### DISCUSSION THREAD

**Discussion by niamtokik** [Kong/insomnia#3633] [Ideas]
Hi everyone,

I am actually using `insomnia-inso` in version 2.2.28, directly from sources, running on ubuntu 20.04 with node 10.19.0. The design document was created using insomnia-designer and OpenAPI 3.0.0 schema. Insomnia-designer was installed directly from the official website, by using snap.

The API I am documenting right know is using authentication to retrieve a token based on JWT. To have access to this token, I need to use a request with valid credentials. So, the idea is the following:

0. Define the security scheme

```yaml
# ...
components:
  securitySchemes:
    BearerToken:
      type: http
      name: Authorization
      in: header
      scheme: bearer
      bearerFormat: bearer
# ...
```

1. Create the request to the JWT authentication. User and password are set in environment.
2. Create another request to fetch an information, like connected user session state on another end-point.
2.1. The Debug request is automatically generated by the previous security schemes defined in the components section.
2.2. Unfortunatelly, the token is automatically configured using a variables `{{bearerToken}}`
2.3.  To automate thing, I created a [chained request](https://support.insomnia.rest/article/43-chaining-requests), extracting automatically the token from the JWT authentication request
2.4. I pasted the resulting chained request in the variable environment like that:

```json
{
  "user": "my_valid_user",
  "password": "my_valid_password",
  "bearerToken": "{% response 'body', 'req_wrk_23bea...', 'b64::...', 'always', 60 %}"
}
``` 

3. when executed in Insomnia-designer, everything work fine
4. when executed in insomnia-inso, the test with `bearerToken` stored as template fail 

```txt
  1) JWT Authentication
       Authenticate:
     Error: No responses for request
      at nj.renderString (/home/user/src/insomnia/packages/insomnia-inso/node_modules/insomnia-send-request/dist/index.js:6083:26)
      at /home/user/src/insomnia/packages/insomnia-inso/node_modules/nunjucks/src/environment.js:39:5
      at RawTask.call (/home/user/src/insomnia/packages/insomnia-inso/node_modules/asap/asap.js:40:19)
      at flush (/home/user/src/insomnia/packages/insomnia-inso/node_modules/asap/raw.js:50:29)
      at process._tickCallback (internal/process/next_tick.js:61:11)

  2) JWT Authentication
       Check User:
     Error: No responses for request
      at nj.renderString (/home/user/src/insomnia/packages/insomnia-inso/node_modules/insomnia-send-request/dist/index.js:6083:26)
      at /home/user/src/insomnia/packages/insomnia-inso/node_modules/nunjucks/src/environment.js:39:5
      at RawTask.call (/home/user/src/insomnia/packages/insomnia-inso/node_modules/asap/asap.js:40:19)
      at flush (/home/user/src/insomnia/packages/insomnia-inso/node_modules/asap/raw.js:50:29)
      at process._tickCallback (internal/process/next_tick.js:61:11)
```

I tried to fix the issues by upgrading to the last nodejs stable version (14.15.3), and have the same result.

**Comment by nijikokun** [comment]
Currently, templating is non-functional with inso. It's on the roadmap, but we haven't gotten around to adding support for it.

**Comment by niamtokik** [comment]
I can probably help you, but I am really bad with node. If you have an idea how to do that, I can probably start working on it.

**Comment by develohpanda** [comment]
Related to https://github.com/Kong/insomnia/issues/3261, but I'll keep this issue open for the description. The lack of template/plugin support is a known limitation as described above.

**Comment by develohpanda** [comment]
Cross-linking: https://github.com/Kong/insomnia/issues/3858
